src/Cryptol/ModuleSystem/Renamer.hs

```diff
@@ -1073,12 +1073,15 @@ instance Rename Bind where
        depsOf (NamedThing (thing n'))
          do mbSig <- traverse renameSchema (bSignature b)
             shadowNames (fst `fmap` mbSig) $
-              do (patEnv,pats') <- renamePats (bParams b)
+              do (patEnv,pats') <- renamePats (bindParams b)
                  -- NOTE: renamePats will generate warnings,
                  -- so we don't need to trigger them again here.
                  e' <- shadowNames' CheckNone patEnv (rnLocated rename (bDef b))
+                 let bParams' = case bParams b of
+                       PatternParams _ -> PatternParams pats'
+                       DroppedParams rng i -> DroppedParams rng i
```

Comment:
I'd consider introducing a `Rename` instance for `BindParams`, as this would allow you to encapsulate the logic of renaming them in that instance.

Reply:
I think this isn't completely straightforward, because we need to also pull out the new naming environment. I can pull this out into a separate function `renameBindParams`, and put it next to the similar functions `renamePats` etc.